cmd/rpcdaemon/commands/graphql_api.go

```diff
@@ -0,0 +1,171 @@
+
+	b, senders, err := api.getBlockWithSenders(ctx, number, tx)
+	if err != nil {
+		return nil, err
+	}
+	if b == nil {
+		return nil, nil
+	}
+
+	chainConfig, err := api.chainConfig(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	receipts, err := api.getReceipts(ctx, tx, chainConfig, b, b.Body().SendersFromTxs())
+	if err != nil {
+		return nil, fmt.Errorf("getReceipts error: %w", err)
+	}
+	result := make([]map[string]interface{}, 0, len(receipts))
+	for _, receipt := range receipts {
+		txn := b.Transactions()[receipt.TransactionIndex]
+		result = append(result, marshalReceipt(receipt, txn, chainConfig, b.HeaderNoCopy(), txn.Hash(), true))
+	}
+
+	getBlockRes, err := api.delegateGetBlockByNumber(tx, b, number, false)
+	if err != nil {
+		return nil, err
+	}
+	getIssuanceRes, err := api.delegateIssuance(tx, b, chainConfig)
+	if err != nil {
+		return nil, err
+	}
+	feesRes, err := api.delegateBlockFees(ctx, tx, b, senders, chainConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	response := map[string]interface{}{}
+	response["block"] = getBlockRes
+	response["issuance"] = getIssuanceRes
+	response["totalFees"] = hexutil.Uint64(feesRes)
+	response["receipts"] = result
+	return response, nil
+}
+
+func (api *GraphQLAPIImpl) getBlockWithSenders(ctx context.Context, number rpc.BlockNumber, tx kv.Tx) (*types.Block, []common.Address, error) {
+	if number == rpc.PendingBlockNumber {
+		return api.pendingBlock(), nil, nil
+	}
+
+	n, hash, _, err := rpchelper.GetBlockNumber(rpc.BlockNumberOrHashWithNumber(number), tx, api.filters)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	block, senders, err := api._blockReader.BlockWithSenders(ctx, tx, hash, n)
+	return block, senders, err
+}
+
+func (api *GraphQLAPIImpl) delegateGetBlockByNumber(tx kv.Tx, b *types.Block, number rpc.BlockNumber, inclTx bool) (map[string]interface{}, error) {
+	td, err := rawdb.ReadTd(tx, b.Hash(), b.NumberU64())
+	if err != nil {
+		return nil, err
+	}
+	additionalFields := make(map[string]interface{})
+	response, err := ethapi.RPCMarshalBlock(b, inclTx, inclTx, additionalFields)
+	if !inclTx {
+		delete(response, "transactions") // workaround for https://github.com/ledgerwatch/erigon/issues/4989#issuecomment-1218415666
+	}
+	response["totalDifficulty"] = (*hexutil.Big)(td)
+	response["transactionCount"] = b.Transactions().Len()
+
+	if err == nil && number == rpc.PendingBlockNumber {
+		// Pending blocks need to nil out a few fields
+		for _, field := range []string{"hash", "nonce", "miner"} {
+			response[field] = nil
+		}
+	}
+
+	// Explicitly drop unwanted fields
+	response["logsBloom"] = nil
+	return response, err
+}
+
+func (api *GraphQLAPIImpl) delegateIssuance(tx kv.Tx, block *types.Block, chainConfig *chain.Config) (internalIssuance, error) {
+	if chainConfig.Ethash == nil {
+		// Clique for example has no issuance
+		return internalIssuance{}, nil
+	}
+
+	minerReward, uncleRewards := ethash.AccumulateRewards(chainConfig, block.Header(), block.Uncles())
```

Comment:
Do HeaderNoCopy

Reply:
Got that part from OTS code too, not used anymore in my code